esp/services/ws_fs/ws_fsBinding.cpp

```diff
@@ -200,9 +200,11 @@ int CFileSpraySoapBindingEx::onGetInstantQuery(IEspContext &context, CHttpReques
 
 IPropertyTree* CFileSpraySoapBindingEx::createPTreeForXslt(double clientVersion, const char* method, const char* dfuwuid)
 {
-    Owned<IEnvironmentFactory> factory = getEnvironmentFactory();
-    Owned<IConstEnvironment> m_constEnv = factory->openEnvironment();
-    Owned<IPropertyTree> pEnvRoot = &m_constEnv->getPTree();
+    Owned<IEnvironmentFactory> factory = getEnvironmentFactoryWithUpdate();
+    Owned<IConstEnvironment> constEnv = factory->openEnvironment();
+    if (!constEnv)
+        throw MakeStringException(ECLWATCH_CANNOT_GET_ENV_INFO, "Failed to get environment information.");
+    Owned<IPropertyTree> pEnvRoot = &constEnv->getPTree();
```

Comment:
should really be trying to avoid this, see other comments (but it's separate issue)